server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMessageListMethod.java

```diff
@@ -125,9 +127,9 @@ private GetMessageListResponse getMessageListResponse(GetMessageListRequest mess
             MultimailboxesSearchQuery searchQuery = convertToSearchQuery(messageListRequest);
             mailboxManager.search(searchQuery,
                 mailboxSession,
-                messageListRequest.getLimit().orElse(maximumLimit) + messageListRequest.getPosition())
+                messageListRequest.getLimit().map(Number::asInt).orElse(maximumLimit) + messageListRequest.getPosition().map(Number::asLong).orElse(DEFAULT_POSITION))
                 .stream()
-                .skip(messageListRequest.getPosition())
+                .skip(messageListRequest.getPosition().map(Number::asLong).orElse(DEFAULT_POSITION))
```

Comment:
why asLong here but asInt just before?

Reply:
I know it does not consistent but the limit is int value and position is long value
Anyway, the parameter of function is long so I changed them all to asLong